satpy/tests/reader_tests/test_oli_tirs_l1_tif.py

```diff
@@ -0,0 +1,576 @@
+    <GRID_CELL_SIZE_THERMAL>30.00</GRID_CELL_SIZE_THERMAL>
+    <ORIENTATION>NORTH_UP</ORIENTATION>
+    <RESAMPLING_OPTION>CUBIC_CONVOLUTION</RESAMPLING_OPTION>
+  </LEVEL1_PROJECTION_PARAMETERS>
+</LANDSAT_METADATA_FILE>
+"""
+
+class TestOLITIRSL1(unittest.TestCase):
+    """Test generic image reader."""
+
+    def setUp(self):
+        """Create temporary images and metadata to test on."""
+        import tempfile
+        from datetime import datetime
+
+        from pyresample.geometry import AreaDefinition
+
+        from satpy.scene import Scene
+
+        self.date = datetime(2024, 5, 12)
+
+        self.filename_info = dict(observation_date=datetime(2024, 5, 3),
+                                  platform_type="L",
+                                  process_level_correction="L1TP",
+                                  spacecraft_id="08",
+                                  data_type="C")
+        self.ftype_info = {"file_type": "granule_b04"}
+
+        # Create area definition
+        pcs_id = "WGS 84 / UTM zone 40N"
+        proj4_dict = {"proj": "utm", "zone": 40, "datum": "WGS84", "units": "m", "no_defs": None, "type": "crs"}
+        self.x_size = 100
+        self.y_size = 100
+        area_extent = (619485., 2440485., 850515., 2675715.)
+        self.area_def = AreaDefinition("geotiff_area", pcs_id, pcs_id,
+                                       proj4_dict, self.x_size, self.y_size,
+                                       area_extent)
+
+        # Create datasets for L, LA, RGB and RGBA mode images
+        self.test_data__1 = da.random.randint(12000, 16000,
+                                              size=(self.y_size, self.x_size),
+                                              chunks=(50, 50)).astype(np.uint16)
+        self.test_data__2 = da.random.randint(8000, 14000,
+                                              size=(self.y_size, self.x_size),
+                                              chunks=(50, 50)).astype(np.uint16)
+        self.test_data__3= da.random.randint(0, 10000,
+                                             size=(self.y_size, self.x_size),
+                                             chunks=(50, 50)).astype(np.uint16)
+
+        ds_b4 = xr.DataArray(self.test_data__1,
+                             dims=("y", "x"),
+                             attrs={"name": "b04",
+                                    "start_time": self.date})
+
+        ds_b11 = xr.DataArray(self.test_data__2,
+                              dims=("y", "x"),
+                              attrs={"name": "b04",
+                                     "start_time": self.date})
+
+        ds_sza = xr.DataArray(self.test_data__3,
+                              dims=("y", "x"),
+                              attrs={"name": "sza",
+                                     "start_time": self.date})
+
+        # Temp dir for the saved images
+        self.base_dir = tempfile.mkdtemp()
```

Comment:
pytest's `tmp_path`is the way to go (gets cleaned up automatically also).

Also we usually have fixtures for test/synthetic files.

Reply:
Unfortunately I don't know how fixtures work, so will be unable to add that. I'll change `tmp_path`though.